pkg/ibm/ibmp.go

```diff
@@ -140,7 +119,55 @@ func (r IBMPowerDynamicConfig) GetInstanceAddress(kubeClient client.Client, ctx
 }
 
 func (r IBMPowerDynamicConfig) ListInstances(kubeClient client.Client, ctx context.Context, instanceTag string) ([]cloud.CloudVMInstance, error) {
-	return nil, fmt.Errorf("not impelemented")
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info("Listing instances by tag:" + instanceTag)
+	service, err := r.authenticate(kubeClient, ctx)
+	if err != nil {
+		return nil, err
+	}
+	builder := core.NewRequestBuilder(core.GET).WithContext(ctx)
+	builder.EnableGzipCompression = service.GetEnableGzipCompression()
+
+	pathParamsMap := map[string]string{
+		"cloud": r.pCloudId(),
+	}
+	_, err = builder.ResolveRequestURL(r.Url, `/pcloud/v1/cloud-instances/{cloud}/pvm-instances`, pathParamsMap)
+	if err != nil {
+		return nil, err
+	}
+	builder.AddHeader("CRN", r.CRN)
+	builder.AddHeader("Accept", "application/json")
+
+	request, err := builder.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	instances := models.PVMInstances{}
```

Comment:
its does not seem like using that models library provides a net improvement for the code

Reply:
Its already sets us free from those manual json manipulation mess and/or writing custom models.
With some more investment it will allow making the API requests out of the box without having to manuallly constructing them.